samcli/lib/remote_invoke/sqs_invoke_executors.py

```diff
@@ -0,0 +1,146 @@
+
+    def _execute_action(self, payload: str) -> RemoteInvokeIterableResponseType:
+        """
+        Calls "send_message" method to send a message to the SQS queue.
+
+        Parameters
+        ----------
+        payload: str
+            The MessageBody which will be sent to the SQS
+
+        Yields
+        ------
+        RemoteInvokeIterableResponseType
+            Response that is consumed by remote invoke consumers after execution
+        """
+        if payload:
+            self.request_parameters[MESSAGE_BODY] = payload
+        else:
+            self.request_parameters[MESSAGE_BODY] = "{}"
+        self.request_parameters[QUEUE_URL] = self._queue_url
+        LOG.debug(
+            "Calling sqs_client.send_message with QueueUrl:%s, MessageBody:%s",
+            self.request_parameters[QUEUE_URL],
+            payload,
+        )
+        try:
+            send_message_response = cast(dict, self._sqs_client.send_message(**self.request_parameters))
+
+            if self._remote_output_format == RemoteInvokeOutputFormat.JSON:
+                yield RemoteInvokeResponse(send_message_response)
+            if self._remote_output_format == RemoteInvokeOutputFormat.TEXT:
+                # Create an object with MD5OfMessageBody and MessageId fields, and write to stdout
+                md5_of_message_body = send_message_response.get("MD5OfMessageBody", "")
+                message_id = send_message_response.get("MessageId", "")
+                md5_of_message_attributes = send_message_response.get("MD5OfMessageAttributes", "")
+                if md5_of_message_body and message_id:
+                    output_data = {"MD5OfMessageBody": md5_of_message_body, "MessageId": message_id}
+                    if md5_of_message_attributes:
+                        output_data["MD5OfMessageAttributes"] = md5_of_message_attributes
+                    yield RemoteInvokeResponse(output_data)
+                    return
+        except ParamValidationError as param_val_ex:
+            raise InvalidResourceBotoParameterException(
+                f"Invalid parameter key provided."
+                f" {str(param_val_ex).replace(f'{QUEUE_URL}, ', '').replace(f'{MESSAGE_BODY}, ', '')}"
+            )
+        except ClientError as client_ex:
+            raise ErrorBotoApiCallException(client_ex) from client_ex
+
+
+def get_queue_url_from_arn(sqs_client: SQSClient, queue_name: str) -> str:
+    """
+    This function gets the queue url of the provided SQS queue name
+
+    Parameters
+    ----------
+    sqs_client: SQSClient
+        SQS client to call boto3 APIs
+    queue_name: str
+        Name of SQS queue used to get the queue_url
+
+    Returns
+    -------
+    str
+        Returns the SQS queue url
+
+    """
+    try:
+        output_response = sqs_client.get_queue_url(QueueName=queue_name)
+        queue_url = cast(str, output_response.get(QUEUE_URL, ""))
```

Comment:
nit: do you still need to cast if you provide default value to `{map}.get` function? I thought return type is implicitly decided by the type of the default value.

Reply:
`mypy` gives an error if it's not cast to a `str` :( `error: Incompatible return value type (got "object", expected "str")  [return-value]`.